src/rubrix/client/datasets.py

```diff
@@ -393,6 +403,48 @@ def _from_pandas(cls, dataframe: pd.DataFrame) -> "DatasetForTextClassification"
             [TextClassificationRecord(**row) for row in dataframe.to_dict("records")]
         )
 
+    @_requires_datasets
+    def prepare_for_training(self) -> "datasets.Dataset":
+        """Prepares the dataset for training.
+
+        This will return a "datasets.Dataset" with two columns: "text" and "label"
+        - Records without an annotation are removed.
+        - The "text" column is taken from the _inputs_ attribute of the records.
+            If there is more than one key in the dictionary, the values will be joined together with white space.
+        - The "label" column corresponds to the _annotation_ attribute of the records.
+        - Labels are transformed to integers.
+
+        Returns:
+            A datasets.Dataset with two columns: "text" and "label".
+        """
+        import datasets
+
+        labels = set()
+        for rec in self._records:
+            annotation = (
+                rec.annotation if isinstance(rec.annotation, list) else [rec.annotation]
+            )
+            if annotation == [None]:
+                continue
+            labels.update(annotation)
+
+        class_label = datasets.ClassLabel(names=sorted(list(labels)))
+        features = datasets.Features(
+            text=datasets.Value("string"),
+            label=[class_label] if self._records[0].multi_label else class_label,
+        )
+
+        text, label = [], []
+        for rec in self._records:
```

Comment:
Would it be possible to only loop through the records once, or reduce the cost of computing labels and text+label?

Reply:
Great! 💯